src/matmul.jl

```diff
@@ -599,11 +599,34 @@ function generic_syrk!(C::StridedMatrix{T}, A::StridedVecOrMat{T}, conjugate::Bo
         throw(DimensionMismatch(lazy"output matrix has size: $(size(C)), but should have size $((mA, mA))"))
     end
 
-    _rmul_or_fill!(C, β)
+    if (!iszero(β) || isempty(A)) # return C*beta
+        _rmul_or_fill!(C, β)
+    else # iszero(β) && A is non-empty
+        if aat
+            for j ∈ 1:m
+                A_1j = A[j,1]'
+                for i ∈ 1:j
+                    A_ij = A[i,1]*A_1j
+                    z1 = zero(A_ij + A_ij)
+                    C[i,j] = convert(promote_type(typeof(z1), eltype(C)), z1)
```

Comment:
why is this necessary? can `C[i,j]` have any type other than `eltype(C)`?

Reply:
This is copied from the "old" most generic matmul code, but since potential promotion to the eltype of `C` is included in the `setindex!` operation here, I guess/agree this could be simply reduced to

```suggestion
                    C[i,j] = zero(A_ij + A_ij)
```

Reply:
I see. Since `generic_syrk` is only written for `<:Number` that is rather overkill. Can't the whole thing be simplified to `fill!(C, zero(abs2(A[1,1]))` then?

Reply:
I'm not sure. In https://github.com/JuliaLang/LinearAlgebra.jl/issues/1486#issuecomment-3570783725, the author says that matrices can be filled with numbers with different moduli, so I think we should keep as much flexibility/genericity here... unless I'm missing something.

Reply:
If you do this you won't be able to multiply them, though.

Reply:
I wasn't sure either, but the author said different moduli are all over the place. I don't know myself (and don't have complicated usecases), I'm just concerned that this is the most generic version, and we should make sure it works under the most generic conditions. I wonder actually why we restricted this to `Number`s in the first place. Gotta find the original PR.

Reply:
We restricted it to `Number` because the only point of this function is speed, there is already a generic fallback for arbitrary types. And it's simpler and faster if we restrict it to `Number`.

I think generic code is good as long as we have a concrete use case, otherwise it's the path to madness.

Reply:
I went basically with your suggestion.

Reply:
I appreciate that. Why didn't you use `fill!`, though?

Reply:
Because we say that we only compute the upper triangle. Perhaps I could use `fill!(UpperTriangular(C), ...)`.